runtime/bcutil/ClassFileOracle.cpp

```diff
@@ -1158,20 +1163,70 @@ ClassFileOracle::walkMethodCodeAttributeAttributes(U_16 methodIndex)
-						U_16 index = localVariableTypeTable->localVariableTypeTable[localVariableTypeTableIndex].index;
-						if (codeAttribute->maxLocals <= index) {
+						UDATA entryFound = FALSE;
+						J9CfrLocalVariableTypeTableEntry lvttEntry = localVariableTypeTable->localVariableTypeTable[localVariableTypeTableIndex];
+						if (codeAttribute->maxLocals <= lvttEntry.index) {
 							Trc_BCU_ClassFileOracle_walkMethodCodeAttributeAttributes_LocalVariableTypeTableIndexOutOfBounds(
-									index, codeAttribute->maxLocals, (U_32)getUTF8Length(_classFile->methods[methodIndex].nameIndex), getUTF8Data(_classFile->methods[methodIndex].nameIndex));
-							_buildResult = GenericError;
+									lvttEntry.index, codeAttribute->maxLocals, (U_32)getUTF8Length(_classFile->methods[methodIndex].nameIndex), getUTF8Data(_classFile->methods[methodIndex].nameIndex));
+							/* throw error */
+							PORT_ACCESS_FROM_PORT(_context->portLibrary());
+							_context->javaVM()->dynamicLoadBuffers->classFileError = (U_8 *) j9mem_allocate_memory(sizeof(J9CfrError), J9MEM_CATEGORY_CLASSES);
+							buildError((J9CfrError*)_context->javaVM()->dynamicLoadBuffers->classFileError, J9NLS_CFR_LVTT_INDEX_OUTOFRANGE__ID, GenericErrorCustom, lvttEntry.index);
+							_buildResult = GenericErrorCustom;
 							break;
-						} else if (NULL == _methodsInfo[methodIndex].localVariablesInfo[index].localVariableTypeTable) {
-							_methodsInfo[methodIndex].localVariablesInfo[index].localVariableTypeTable = localVariableTypeTable;
-						} else if (localVariableTypeTable != _methodsInfo[methodIndex].localVariablesInfo[index].localVariableTypeTable) {
+						} else if (NULL == _methodsInfo[methodIndex].localVariablesInfo[lvttEntry.index].localVariableTypeTable) {
+							_methodsInfo[methodIndex].localVariablesInfo[lvttEntry.index].localVariableTypeTable = localVariableTypeTable;
+						} else if (localVariableTypeTable != _methodsInfo[methodIndex].localVariablesInfo[lvttEntry.index].localVariableTypeTable) {
 							Trc_BCU_ClassFileOracle_walkMethodCodeAttributeAttributes_DuplicateLocalVariableTypeTable(
-									(U_32)getUTF8Length(_classFile->methods[methodIndex].nameIndex), getUTF8Data(_classFile->methods[methodIndex].nameIndex), localVariableTypeTable, _methodsInfo[methodIndex].localVariablesInfo[index].localVariableTypeTable);
+									(U_32)getUTF8Length(_classFile->methods[methodIndex].nameIndex), getUTF8Data(_classFile->methods[methodIndex].nameIndex), localVariableTypeTable, _methodsInfo[methodIndex].localVariablesInfo[lvttEntry.index].localVariableTypeTable);
 							_buildResult = GenericError;
 							break;
 						}
+
+						/* Verify that a matching entry exists in a LocalVariableTable entry. */
+						if  (_methodsInfo[methodIndex].localVariablesInfo[lvttEntry.index].localVariableTable != NULL ) {
+							/* Check to see if an entry at the lvtt index has been recorded. This should be the common case.
```

Comment:
Better change to NULL != _methodsInfo[methodIndex].localVariablesInfo[lvttEntry.index].localVariableTable